src/Views/Homepage.vala

```diff
@@ -98,11 +99,62 @@ namespace AppCenter {
                 new Thread<void*> ("update-banner", run);
             });
 
+            var recently_updated_label = new Gtk.Label (_("Recently Updated"));
+            recently_updated_label.get_style_context ().add_class ("h4");
+            recently_updated_label.xalign = 0;
+            recently_updated_label.margin_start = 10;
+
+            var recently_updated_carousel = new Widgets.Carousel ();
+
+            var recently_updated_grid = new Gtk.Grid ();
+            recently_updated_grid.margin = 2;
+            recently_updated_grid.margin_top = 12;
+            recently_updated_grid.attach (recently_updated_label, 0, 0, 1, 1);
+            recently_updated_grid.attach (recently_updated_carousel, 0, 1, 1, 1);
+
+            var recently_updated_revealer = new Gtk.Revealer ();
+            recently_updated_revealer.add (recently_updated_grid );
+
+            houston.get_updated.begin ((obj, res) => {
+                var updated_ids = houston.get_updated.end (res);
+                List<AppCenterCore.Package> packages_for_carousel = new List<AppCenterCore.Package> ();
+                ThreadFunc<void*> run = () => {
+                    uint packages_added = 0;
+                    foreach (var package in updated_ids) {
+                        if (packages_added >= NUM_PACKAGES_IN_CAROUSEL) {
+                            break;
+                        }
+
+                        var candidate = package + ".desktop";
+                        var candidate_package = AppCenterCore.Client.get_default ().get_package_for_component_id (candidate);
+
+                        if (candidate_package != null) {
+                            candidate_package.update_state ();
+                            if (candidate_package.state == AppCenterCore.Package.State.NOT_INSTALLED) {
+                                packages_added++;
+                                packages_for_carousel.append (candidate_package);
+                            }
+                        }
+                    }
+                    
+                    Idle.add (() => {
+                        foreach (var banner_package in packages_for_carousel) {
+                            recently_updated_carousel.add_package (banner_package);
+                        }
+                        recently_updated_revealer.reveal_child = true;
+                        main_window.homepage_loaded ();
+                        return false;
+                    });
+                    return null;
+                };
+                new Thread<void*> ("update-banner", run);
```

Comment:
Can we rename this thread so that it doesn't match the other thread for the banner? Only really useful if we ever need to differentiate the two threads during debugging, but worth it nontheless.